index.js

```diff
@@ -59,14 +62,16 @@ async function create ({
       })
     }
 
-    let hasUploaded = false
+    await once(metadata, 'peer-open')
 
-    metadata.on('upload', () => {
-      hasUploaded = true
-    })
+    // bitfield-like structure. Used to known start and ends of all the hyperdrive contents.
+    // Later we can use it to compare what other peers have.
+    const [, localBitfield, visited] = await getFileRanges(drive)
 
-    await once(metadata, 'peer-open')
+    await peersFullAck(localBitfield, visited)
```

Comment:
We should start listening on the `peer-ack` event right after `ready` so that we don't miss any of the events.

Reply:
yeah! you are right. Also I missed to pass the content feed at the beginning I think. 

I'm worried about missing 'peer-ack' events